plugins/modules/instance.py

```diff
@@ -372,6 +372,12 @@
             'have status "running".'
         ],
     ),
+    additional_ipv4=SpecField(
```

Comment:
We can use the `choices` field here to only allow `public` or `private`. 

I also wonder if this could be made a bit more future-proof by making this field a list of dicts. This might come in handy if we need configuration fields in the future 👍 

e.g. 
```
additional_ipv4:
  - public: true
  - public: false
```

Take a look at the [firewall module](https://github.com/linode/ansible_linode/blob/dev/plugins/modules/firewall.py#L134) for a good example of how the spec entry would look.

Reply:
```
additional_ipv4:
  - public: true
  - public: false
```

Does something like this makes it have 2 additional IP addresses?